src/main/java/org/prebid/server/auction/CurrencyService.java

```diff
@@ -0,0 +1,128 @@
+package org.prebid.server.auction;
+
+import org.apache.commons.collections4.MapUtils;
+import org.prebid.server.auction.model.Currency;
+import org.prebid.server.exception.PreBidException;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Service for price currency conversion between currencies.
+ */
+public class CurrencyService {
+
+    private static final Integer SCALE = 5;
+    private static final String DEFAULT_BID_CURRENCY = "USD";
+
+    private final LatestRatesService latestRatesService;
+
+    public CurrencyService(LatestRatesService latestRatesService) {
+        this.latestRatesService = latestRatesService;
+    }
+
+    /**
+     * Converts price from bidCurrency to adServerCurrency using rates defined in request or if absent, from
+     * latest service currency. Throws {@link PreBidException} in case conversion is not possible.
+     */
+    public BigDecimal convertCurrency(BigDecimal price, Currency requestCurrencyRates, String adServerCurrency,
+                                      String bidCurrency) {
+        if (Objects.equals(adServerCurrency, bidCurrency)) {
+            return price;
+        }
+
+        // use Default USD currency if bidder left this field empty. After, when bidder will implement multi currency
+        // support it will be changed to throwing PerbidException.
+        if (bidCurrency == null) {
+            bidCurrency = DEFAULT_BID_CURRENCY;
+        }
+
+        // get conversion rate from request currency rates if it is present
+        BigDecimal conversionRate = null;
+        if (requestCurrencyRates != null) {
+            conversionRate = getConversionRate(requestCurrencyRates.getConversions(), adServerCurrency, bidCurrency);
+        }
+
+        // if conversion rate from requestCurrency was not found, try the same from latest currencies
+        if (conversionRate == null) {
+            final Map<String, Map<String, BigDecimal>> latestRates = latestRatesService.getRates();
+
+            if (latestRates != null) {
```

Comment:
Perhaps, it's a redundant check in case it also checked inside _CurrencyService#getConversionRate_